sensors/depth/depth.go

```diff
@@ -0,0 +1,42 @@
+// Package depth implements the Depth sensor
+package depth
+
+import (
+	"context"
+
+	"github.com/pkg/errors"
+	"go.viam.com/rdk/components/camera"
+	"go.viam.com/rdk/registry"
+
+	"go.viam.com/slam/sensors/utils"
+)
+
+// Depth represents a depth sensor.
+type Depth struct {
+	Name  string
+	depth camera.Camera
+}
+
+// New creates a new Depth sensor based on the sensor definition and the service config.
+func New(ctx context.Context, deps registry.Dependencies, sensors []string, sensorIndex int) (Depth, error) {
```

Comment:
[note to self] This New function (and the ones for lidar and rgb) will approximately replace [this call](https://github.com/viamrobotics/rdk/blob/b4a5473abe9a4721ddfc5300c87927e52c7511fb/services/slam/builtin/builtin.go#L491), though builtin.go itself will be replaced with slam service implementations in the carto and orbslam repos.